src/components/FormElements/XdrLedgerKeyPicker.tsx

```diff
@@ -0,0 +1,406 @@
+
+  const isXdrInit = useIsXdrInit();
+
+  const xdrDecodeJson = () => {
+    if (!isXdrInit) {
+      return null;
+    }
+
+    try {
+      const xdrJson = StellarXdr.decode("LedgerKey", value);
+
+      return {
+        jsonString: xdrJson,
+        error: "",
+      };
+    } catch (e) {
+      return {
+        jsonString: "",
+        error: `Unable to decode input as LedgerKey: ${e}`,
+      };
+    }
+  };
+
+  const jsonEncodeXdr = (str: string) => {
+    if (!(isXdrInit && str && "LedgerKey")) {
+      return null;
+    }
+
+    try {
+      JSON.parse(str);
+    } catch (e) {
+      return {
+        xdrString: "",
+        error: "Invalid JSON",
+      };
+    }
+
+    try {
+      const xdrString = StellarXdr.encode("LedgerKey", str);
+
+      return {
+        xdrString: xdrString,
+        error: "",
+      };
+    } catch (e) {
+      return {
+        xdrString: "",
+        error: `Unable to decode JSON as LedgerKey: ${e}`,
+      };
+    }
+  };
+
+  useEffect(() => {
+    if (ledgerKeyXdr) {
+      onChange(ledgerKeyXdr);
+    }
+  }, [ledgerKeyXdr]);
+
+  const getKeyType = (val: string) =>
+    ledgerKeyFields.find((field) => field.id === val);
+
+  const xdrJsonDecoded = xdrDecodeJson();
+
+  useEffect(() => {
+    if (xdrJsonDecoded?.jsonString) {
+      setLedgerKeyJsonString(xdrJsonDecoded.jsonString);
+
+      const xdrJsonDecodedJSON = JSON.parse(xdrJsonDecoded.jsonString);
+      const decodedKeyType = Object.keys(xdrJsonDecodedJSON)[0];
+      const selectedKeyType = getKeyType(decodedKeyType);
+
+      if (selectedKeyType) {
+        selectLedgerKey(selectedKeyType);
+      }
+    }
+  }, [xdrJsonDecoded?.jsonString]);
+
+  useEffect(() => {
+    if (value && xdrJsonDecoded?.error) {
+      setLedgerKeyXdrError(xdrJsonDecoded?.error);
+    } else {
+      setLedgerKeyXdrError("");
+    }
+  }, [xdrJsonDecoded?.error]);
+
+  useEffect(() => {
+    if (ledgerKeyXdr) {
+      const error = validate.getXdrError(ledgerKeyXdr, "LedgerKey");
+
+      if (error?.result === "error") {
+        setLedgerKeyXdrError(error.message);
+      } else {
+        setLedgerKeyXdrError("");
+      }
+    } else {
+      setLedgerKeyXdrError("");
+    }
+  }, [ledgerKeyXdr]);
+
+  useEffect(() => {
+    if (!ledgerKeyJsonString && selectedLedgerKey) {
+      const templatesArr = selectedLedgerKey.templates.split(",");
+      const formLedgerKeyJson = templatesArr.reduce((accr, item) => {
+        accr[selectedLedgerKey.id] = {
+          ...accr[selectedLedgerKey.id],
+          [item]: "",
+        };
+
+        return accr;
+      }, {} as AnyObject);
+
+      if (selectedLedgerKey.addlFields) {
+        formLedgerKeyJson[selectedLedgerKey.id] = selectedLedgerKey.addlFields;
+      }
+
+      const formLedgerKeyJsonString = JSON.stringify(formLedgerKeyJson);
+      setLedgerKeyJsonString(formLedgerKeyJsonString);
+    }
+  }, [ledgerKeyJsonString, selectedLedgerKey]);
```

Comment:
We're doing a lot in `useEffect` in a single component. This might affect performance if we're not careful. Maybe we can simplify these?

I think it's not good to have a `useEffect` dependency that can go from `undefined` to a value. It would be better to use a variable. For example:

```typescript
// This way, we always check a string
const decodedJsonString = xdrJsonDecoded?.jsonString || "";
```